src/gmp/models/portlist.ts

```diff
@@ -0,0 +1,121 @@
+/* SPDX-FileCopyrightText: 2024 Greenbone AG
+ *
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import Model, {
+  Element,
+  ModelElement,
+  ModelProperties,
+  parseModelFromElement,
+} from 'gmp/model';
+import {parseInt, parseBoolean} from 'gmp/parser';
+import {map} from 'gmp/utils/array';
+import {isDefined} from 'gmp/utils/identity';
+
+export type ProtocolType = 'tcp' | 'udp';
+
+interface PortRangeElement extends ModelElement {
+  type: ProtocolType;
+  port_list_id?: string;
+  start: number;
+  end: number;
+  [key: string]: unknown;
+}
+
+interface PortRangeProperties extends ModelProperties {
+  protocol_type: ProtocolType;
+  port_list_id: string;
+  start: number;
+  end: number;
+}
+
+export interface PortListElement extends ModelElement {
+  id?: string;
+  port_ranges?: {
+    port_range: PortRangeElement[];
+  };
+  port_count?: {
+    all?: string;
+    tcp?: string;
+    udp?: string;
+  };
+  targets?: {
+    target: ModelElement[];
+  };
+  predefined?: string;
+}
+
+interface PortCount {
+  all: number;
+  tcp: number;
+  udp: number;
+}
+
+interface PortListProperties extends ModelProperties {
+  port_ranges?: PortRangeProperties[];
+  port_count: {
+    all: number;
+    tcp: number;
+    udp: number;
+  };
+  targets?: Model[];
+  predefined?: boolean;
+}
+
+export class PortRange extends Model {
+  static entityType = 'portrange';
+  protocol_type!: ProtocolType;
+  port_list_id!: string;
+  start!: number;
+  end!: number;
+
+  parseProperties(element: PortRangeElement): PortRangeProperties {
+    return PortRange.parseElement(element);
+  }
+
+  static parseElement(element: PortRangeElement): PortRangeProperties {
+    const ret = super.parseElement(element) as PortRangeProperties;
+    ret.protocol_type = element.type;
+    return ret;
+  }
+}
+
+class PortList extends Model {
+  static entityType = 'portlist';
+  port_ranges!: PortRange[];
+  port_count!: PortCount;
+  targets!: Model[];
+  predefined!: boolean;
+
+  static parseElement(element: PortListElement): PortListProperties {
+    const ret = super.parseElement(element) as PortListProperties;
+
+    const ranges =
+      isDefined(ret.port_ranges) && isDefined(element.port_ranges)
+        ? element.port_ranges.port_range
+        : [];
+
+    ret.port_ranges = map(ranges, (range: PortRangeElement) => {
+      range.port_list_id = ret.id;
+      return PortRange.fromElement(range as Element) as PortRange;
```

Comment:
May not need as PortRange
```suggestion
      return PortRange.fromElement(range as Element);
```